frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -129,7 +129,7 @@ export function InstantSearchProvider({
             return window.location;
          },
          createURL({ qsModule, routeState, location }) {
-            const baseUrl = location.origin;
```

Comment:
Can we omit baseUrl and let these be relative urls?

Reply:
If we choose to make the url returned by this method relative, also consider that this might cause the urls created by calls to `createURL` elsewhere in the app (links below) might also become relative. Maybe that's fine, just wanted to point it out.

https://github.com/iFixit/react-commerce/blob/206ff8be780f671407ccf096500e21823d3be6b4/frontend/components/product-list/sections/FilterableProductsSection/RefinementMenu.tsx#L93-L101

https://github.com/iFixit/react-commerce/blob/959d07417c8016352bd507cb0e7b510028fb6a00/frontend/components/product-list/sections/FilterableProductsSection/Pagination.tsx#L33

Reply:
I like this idea, but it looks trickier to implement than I'd like.

Looking at the code for those two places, they both rely on `URL` to parse apart the URL, in order to modify it. `URL` [does not support](https://github.com/whatwg/url/issues/531) relative URLs, so we would have to find an alternative technique. If I could be sure `createURL` would never return a search string, I could just make the assumption that we only ever get a path back from it (we're already ignoring fragments). But that seems like a risk longer-term, because it violates the implied interface of a method named `createURL`: the assumption is that we get a URL when calling it.

Further, the argument about the name `createURL` also implies that we should always return a complete URL from it.

Given all that, I think I'd like to keep `createURL` returning absolute URLs. What I wouldn't mind doing is dropping the `origin` part of the URL from the `RefinementMenu` use, which should result in relative URLs being rendered for the case of interest.

Reply:
Talked to @sterlinghirsh, and decided on [9cdd74d](https://github.com/iFixit/react-commerce/pull/505/commits/9cdd74d680439be8b2392fde7fb259f4aaf59445).